### PR TITLE
Add log in/out feature

### PR DESCRIPTION
**Pull request by yjzhang111**
Add a logout button when the user is logged in (and a log in button when the user isn't), and automatically brings the user to the login page when trying to leave comment without logging in.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -87,9 +89,14 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     // If statement determines whether the POST request is
     // from limiting a certain number of comments to display
     // or from leaving a new comment
+    UserService userService = UserServiceFactory.getUserService();
     if (request.getParameter("numComment") != null) {
       numComment = Integer.parseInt(request.getParameter("numComment"));
     } else {
+      if (!userService.isUserLoggedIn()) {
```
Comment (albert-g3) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:96]: Does this mean user can enter number of comments to display while logged out?

File: portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+    boolean isLoggedIn;
+
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      isLoggedIn = true;
+    } else {
+      isLoggedIn = false;
+    }
```
Comment (albert-g3) [inline portfolio/src/main/java/com/google/sps/servlets/LoginServlet.java]: Can this be simplified to `boolean isLoggedIn = userService.isUserLoggedIn()`?

File: portfolio/src/main/java/com/google/sps/servlets/UsersServlet.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+public class UsersServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html");
+
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      String urlToRedirectToAfterUserLogsOut = "/";
```
Comment (albert-g3) [inline portfolio/src/main/java/com/google/sps/servlets/UsersServlet.java]: It's good to have descriptive variable names, but also not too long.
I suggest `redirectUrlAfterLogOut`

File: portfolio/src/main/java/com/google/sps/servlets/UsersServlet.java [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+
+      response.getWriter().println("<p>Hello " + userEmail + "!</p>");
+      response.getWriter().println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
+    } else {
+      String urlToRedirectToAfterUserLogsIn = "/";
```
Comment (albert-g3) [inline portfolio/src/main/java/com/google/sps/servlets/UsersServlet.java]: Similar to line 35, try to shorten this

**Review by albert-g3 — APPROVED** [review]
Just minor comments, please take a look

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (yjzhang111) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:96]: Yes, users should be able to see comments and choose the number they want to display even if they're not logged in.

**Pull request merged**